src/map/script.c

```diff
@@ -28628,6 +28628,107 @@ static BUILDIN(mestipbox)
 	return true;
 }
 
+
+/**
+ * Returns a units <parameter>'s values
+ *
+ * NOTE: UNIT_PARAM_MAX_HP needs two arrays.
+ *
+ * getunitparam(<param>{, <class id>{, <maxlv array>, <value array>}})
+ */
+static BUILDIN(getunitparam)
+{
+	int class = -1;
+	if (script_hasdata(st, 3)) {
+		class = script_getnum(st, 3);
+		if (class != -1) {
+			if (!pc->db_checkid(class)) {
+				ShowError("buildin_getunitparam: invalid class (%d)\n", class);
+				st->state = END;
+				return false;
+			}
+			class = pc->class2idx(class);
+		}
+	}
+
+	struct map_session_data *sd = NULL;
+	if (class == -1) {
+		sd = script_rid2sd(st);
+		if (sd == NULL) {
+			ShowError("buildin_getunitparam: No player attached, but class == -1.\n");
+			return false;
+		}
+		class = pc->class2idx(sd->status.class);
+	}
+
+	struct s_unit_params *entry = status->dbs->unit_params[class];
+	int param = script_getnum(st, 2);
+	switch (param) {
+	case UNIT_PARAM_NAME:
+		script_pushconststr(st, entry->name);
+		break;
+	case UNIT_PARAM_NATHEAL_WEIGHT_RATE:
+		script_pushint(st, entry->natural_heal_weight_rate);
+		break;
+	case UNIT_PARAM_MAX_ASPD:
+		script_pushint(st, (2000 - entry->max_aspd) / 10); // max_aspd is actually min_amotion :)
+		break;
+	case UNIT_PARAM_MAX_HP: {
+		if (!script_hasdata(st, 4) || !script_hasdata(st, 5)) {
+			ShowError("buildin_getunitparam: UNIT_PARAM_MAXP_HP requires 4 parameters: <param>, <class id>, <maxlv array>, <value array>\n");
+			st->state = END;
+			return false;
+		}
+		struct script_data *maxhp_maxlvls = script_getdata(st, 4);
+		struct script_data *maxhp_values = script_getdata(st, 5);
+		const char *maxhp_maxlvls_varname = reference_getname(maxhp_maxlvls);
+		const char *maxhp_values_varname = reference_getname(maxhp_values);
```

Comment:
Calling `reference_getname()` on something that is not a reference could cause a segfault (invalid array index accessed). Please ensure that `data_isreference()` before calling this

Reply:
fixed and rebased